src/DataSource.ts

```diff
@@ -204,39 +211,223 @@ function parseTraceResponse(response: DataFrame): DataFrame {
+          data: e,
+        };
+      });
+
+      acc.links.push(...links);
+
+      const node = {
+        name: service.Name,
+        id: service.ReferenceId,
+        data: service,
+      };
+      acc.nodes = {
+        ...acc.nodes,
+        [node.id]: node,
+      };
+
+      return acc;
+    },
+    { nodes: {}, links: [] }
+  );
+  const nodesArray = Object.values(nodes);
+
+  return [
+    new MutableDataFrame({
+      name: 'ServiceMap_services',
+      fields: [
+        {
+          name: 'id',
+          type: FieldType.string,
+          values: new ArrayVector(nodesArray.map((n: any) => n.id)),
+        },
+        {
+          name: 'name',
+          type: FieldType.string,
+          values: new ArrayVector(nodesArray.map((n: any) => n.name)),
+        },
+        {
+          name: 'type',
+          type: FieldType.string,
+          values: new ArrayVector(nodesArray.map((n: any) => n.data.Type)),
+        },
+        {
+          name: 'data',
+          type: FieldType.other,
+          values: new ArrayVector(nodesArray.map((n: any) => n.data)),
+        },
+      ],
+      meta: {
+        // TODO: needs new grafana/data
+        // @ts-ignore
+        preferredVisualisationType: 'serviceMap',
+      },
+    }),
+    new MutableDataFrame({
+      name: 'ServiceMap_edges',
+      fields: [
+        {
+          name: 'source',
+          type: FieldType.string,
+          values: new ArrayVector(links.map((l: any) => l.source)),
+        },
+        {
+          name: 'sourceName',
+          type: FieldType.string,
+          values: new ArrayVector(links.map((l: any) => nodes[l.source].name)),
+        },
+        {
+          name: 'target',
+          type: FieldType.string,
+          values: new ArrayVector(links.map((l: any) => l.target)),
+        },
+        {
+          name: 'targetName',
+          type: FieldType.string,
+          values: new ArrayVector(links.map((l: any) => nodes[l.target].name)),
+        },
+        {
+          name: 'data',
+          type: FieldType.other,
+          values: new ArrayVector(links.map((l: any) => l.data)),
+        },
+      ],
+      meta: {
+        // TODO: needs new grafana/data
+        // @ts-ignore
+        preferredVisualisationType: 'serviceMap',
+      },
+    }),
+  ];
+}
+
+function parseServiceMapResponse(
+  response: DataFrame,
+  instanceSettings: DataSourceInstanceSettings,
+  query?: XrayQuery
+): DataFrame[] {
+  const [servicesFrame, edgesFrame] = parseGraphResponse(response);
+  const serviceQuery = 'service(id(name: "${__data.fields.name}", type: "${__data.fields.type}"))';
+  servicesFrame.fields[0].config = {
+    links: [
+      {
+        title: 'All Traces',
+        url: '',
+        internal: {
+          query: {
+            ...(query || {}),
+            queryType: XrayQueryType.getTraceSummaries,
+            query: serviceQuery,
+          },
+          datasourceUid: instanceSettings.uid,
+          // @ts-ignore
```

Comment:
why the need for ts-ignore for this on?

Reply:
I just have to update the grafana-data dependency I think we still uses the 7.2 version of those here.